src/utils/formatting/content_split.rs

```diff
@@ -148,45 +148,112 @@ const MIN_FREE_CHARS: usize = 2;
 /// Otherwise, we simply return the current line and basically don't do anything.
 fn check_if_full(lines: &mut Vec<String>, content_width: usize, current_line: String) -> String {
     // Already complete the current line, if there isn't space for more than two chars
-    if current_line.width() > content_width.saturating_sub(MIN_FREE_CHARS) {
+    if console::measure_text_width(&current_line) > content_width.saturating_sub(MIN_FREE_CHARS) {
         lines.push(current_line);
         return String::new();
     }
 
     current_line
 }
 
-/// Splits a long word at a given character width.
+const ANSI_RESET : &str = "\u{1b}[0m";
+
+/// Splits a long word at a given character width. Inserting the needed ansi codes to preserve style.
 /// This needs some special logic, as we have to take multi-character UTF-8 symbols into account.
 /// When simply splitting at a certain char position, we might end up with a string that's has a
 /// wider display width than allowed.
 fn split_long_word(allowed_width: usize, word: &str) -> (String, String) {
-    let mut current_width = 0;
-    let mut splitted = String::new();
-
-    let mut char_iter = word.chars().peekable();
-    // Check if the string might be too long, one character at a time.
-    // Peek into the next char and check the exit condition.
-    // That is, pushing the next character would result in the string being too long.
-    while let Some(c) = char_iter.peek() {
-        if (current_width + c.width().unwrap_or(1)) > allowed_width {
-            break;
+    let mut iter = console::AnsiCodeIterator::new(word);
+    let mut current_len = 0;
+
+    let mut escapes = Vec::new();
+    let mut head = String::with_capacity(word.len());
+    let mut last_txt = 0;
+    let mut escape_count = 0;
+    let mut tail = String::with_capacity(word.len());
```

Comment:
Please write proper inline documentation on what this algorithm does.
It's by no means trivial and really hard to read/understand right now.

Try to chunk the algorithm into logical blocks and explain what each block is doing in a sentence or two.

Also, please write docs on what these variables are for.